src/collection-view.js

```diff
@@ -505,17 +502,24 @@ const CollectionView = Backbone.View.extend({
 
   // Internal Method. Add the view to children and render it at the given index.
   _addChildView(view, index) {
-    monitorViewEvents(view);
-
-    // set up the child view event forwarding
-    this._proxyChildEvents(view);
+    this._setupChildView(view, index);
+    this.triggerMethod('before:add:child', this, view);
```

Comment:
I think `before:add:child` should be the first thing to happen in the function before anything else occurs.

Also I don't think there's a need for `_addChildView` we can just put it all in `addChildView`?